datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -563,13 +563,42 @@ pub fn array_slice(args: &[ArrayRef]) -> Result<ArrayRef> {
     define_array_slice(list_array, key, extra_key, false)
 }
 
+fn general_array_pop(
+    list_array: &GenericListArray<i32>,
+    from_back: bool,
+) -> Result<(Vec<i64>, Vec<i64>)> {
+    if from_back {
+        let key = vec![0; list_array.len()];
+        let extra_key: Vec<_> = list_array
+            .iter()
+            .map(|x| x.map_or(0, |arr| arr.len() as i64 - 1))
+            .collect();
+        Ok((key, extra_key))
+    } else {
+        let key: Vec<_> = list_array.iter().map(|x| x.map_or(0, |_arr| 2)).collect();
```

Comment:
what is 2 here? and we dont use `arr` we probably can just put `||`

Reply:
2 is the first element position (position = index + 1) we want in the new array.

We only handle two cases of the first element index: if the old array has any elements, starts from 2 (index + 1), or starts from initial.

Reply:
thanks, please leave the comment in the code what the 2 is, so for other devs it will be easier to read the code.

Reply:
Fixed, thanks again! : )